internal/engine/buildcontroller.go

```diff
@@ -32,23 +33,73 @@ func NewBuildController(b BuildAndDeployer) *BuildController {
 	}
 }
 
+// Algorithm to choose a manifest to build next.
+func nextManifestToBuild(state store.EngineState) model.ManifestName {
+	// First, go through all the manifests in order.
+	// If any of them haven't started yet, build them now.
+	for _, mn := range state.ManifestDefinitionOrder {
+		ms, ok := state.ManifestStates[mn]
+		if ok && !ms.StartedFirstBuild {
+			return mn
+		}
+	}
+
+	// Next go through all the manifests, and check:
+	// 1) all pending file changes, and
+	// 2) all pending manifest changes
+	// The earliest one is the one we want.
+	choiceName := model.ManifestName("")
+	earliest := time.Now()
+
+	// always use a stable iteration order
+	for _, mn := range state.ManifestDefinitionOrder {
+		ms, ok := state.ManifestStates[mn]
+		if !ok {
+			continue
+		}
+
+		// Always prioritize builds that are crash-rebuilding.
+		if ms.CrashRebuildInProg {
```

Comment:
hmm, with this slightly different flow i might rename this flag -- `NeedsRebuildFromCrash` or something? this makes it seem like a pod state (in the old world, this was the state of the MANIFEST)

Reply:
changed it to NeedsRebuildFromCrash

i'm a little bit confused by your comment "in the old world, this was the state of the MANIFEST"...it's always been on the ManifestState and still is afaict?

Reply:
i... am no longer sure what I meant by that comment, ignore 😅